src/BlockTypes/ProductStockIndicator.php

```diff
@@ -48,7 +50,88 @@ protected function get_block_type_supports() {
 	 * This registers the scripts; it does not enqueue them.
 	 */
 	protected function register_block_type_assets() {
-		parent::register_block_type_assets();
-		$this->register_chunk_translations( [ $this->block_name ] );
+		return null;
+	}
+
+	/**
+	 * Register the context.
+	 */
+	protected function get_block_type_uses_context() {
+		return [ 'query', 'queryId', 'postId' ];
+	}
+
+	/**
+	 * Get stock text based on stock. For example:
+	 * - In stock
+	 * - Out of stock
+	 * - Available on backorder
+	 * - 2 left in stock
+	 *
+	 * @param [bool]     $is_in_stock Whether the product is in stock.
+	 * @param [bool]     $is_low_stock Whether the product is low in stock.
+	 * @param [int|null] $low_stock_amount The amount of stock that is considered low.
+	 * @param [mixed]    $is_on_backorder Whether the product is on backorder.
```

Comment:
Isn't `$is_on_backorder` a bool?

Reply:
Nice catch! Thanks for pointing this out 🙏